libkbfs/tlf_journal.go

```diff
@@ -476,25 +517,33 @@ func (j *tlfJournal) doBackgroundWorkLoop(
 				j.log.CDebugf(ctx,
 					"Got pause signal for %s", j.tlfID)
 				bws = TLFJournalBackgroundWorkPaused
+				j.pause(journalPauseCommand)
 
 			case <-j.needShutdownCh:
 				j.log.CDebugf(ctx,
 					"Got shutdown signal for %s", j.tlfID)
 				needShutdown = true
 			}
 
-			errCh = nil
 			// Cancel the worker goroutine as we exit this
 			// state.
 			bwCancel()
 			bwCancel = nil
+
+			// Ensure the worker finishes after being canceled, so it
+			// doesn't pick up any new work.  For example, if the
+			// worker doesn't check for cancellations before checking
+			// the journal for new work, it might process some journal
+			// entries before returning an error.
+			<-errCh
+			errCh = nil
```

Comment:
I think there's a comment in `shutdown` that doesn't apply anymore (about the background goroutine finishing first)